query-engine/core/src/query_graph_builder/write/nested/upsert_nested.rs

```diff
@@ -169,7 +169,27 @@ pub fn nested_upsert(
             ),
         )?;
 
-        graph.create_edge(&if_node, &update_node, QueryGraphDependency::Then)?;
+        // In case the connector doesn't support referential integrity, we add a subtree to the graph that emulates the ON_UPDATE referential action.
```

Comment:
I know it's tedious, but it's important that we keep the top level graphs consistent with the code. Make sure you get a Monodraw license from the company and I will provide you the file I'm working with right now so you can update the graph drawing.